contracts/SpokePool.sol

```diff
@@ -938,72 +949,69 @@ abstract contract SpokePool is
      *         USS RELAYER FUNCTIONS          *
      ******************************************/
 
-    function fillUSSRelay(
-        address depositor,
-        address recipient,
-        address exclusiveRelayer,
-        InputToken memory inputToken,
-        OutputToken memory outputToken,
-        uint256 repaymentChainId,
-        uint256 originChainId,
-        uint32 depositId,
-        uint32 fillDeadline,
-        bytes memory message
-    ) public override nonReentrant unpausedFills {
-        // Validate input params
+    function fillUSSRelay(USSRelayData memory relayData, uint256 repaymentChainId)
+        public
+        override
+        nonReentrant
+        unpausedFills
+    {
+        // Exclusivity deadline is inclusive and is the latest timestamp that the exclusive relayer has sole right
+        // to fill the relay.
+        // solhint-disable-next-line not-rely-on-time
+        if (relayData.exclusiveRelayer != msg.sender && relayData.exclusivityDeadline >= block.timestamp) {
+            revert NotExclusiveRelayer();
+        }
 
-        USSRelayExecution memory relayExecution = USSRelayExecution({
-            relay: USSRelayData({
-                depositor: depositor,
-                recipient: recipient,
-                relayer: exclusiveRelayer,
-                inputToken: inputToken.token,
-                outputToken: outputToken.token,
-                inputAmount: inputToken.amount,
-                outputAmount: outputToken.amount,
-                originChainId: originChainId,
-                destinationChainId: chainId(),
-                depositId: depositId,
-                fillDeadline: fillDeadline,
-                message: message
-            }),
+        // No need to override passed in destination chain id to be equal to chainId(),
+        // since this is not emitted in the FilledRelay event. So we do not have to fear a replay attack.
```

Comment:
I don't quite understand this? What has changed that allows us to be less worried about replay attacks or people spoofing the chain id?

Reply:
the point I'm trying to make here is that `destinationChainId` isn't used in `_fillRelay(USS)` and its also not emitted in `FilledUSSRelay`, but I think I'm overlooking the fact actually that its part of the relayHash, oops

Reply:
015c116193aa1c716a5e4f81035712395fce5235